osu.Game/Rulesets/Difficulty/Skills/StrainSkill.cs

```diff
@@ -57,6 +58,14 @@ public sealed override void Process(DifficultyHitObject current)
                 saveCurrentPeak();
                 startNewSectionFrom(currentSectionEnd, current);
                 currentSectionEnd += SectionLength;
+
+                // If the current peak is 0 it can't decay any further so we can leave early
+                if (Precision.AlmostEquals(currentSectionPeak, 0) && current.StartTime > currentSectionEnd)
```

Comment:
Unsure about the preferred acceptable difference for double comparison so I've kept it at the default for now.